alvr/dashboard/src/dashboard/components/settings_controls/mod.rs

```diff
@@ -51,6 +51,23 @@ pub fn get_display_name(id: &str, strings: &HashMap<String, String>) -> String {
     })
 }
 
+pub fn f64_eq(f1: f64, f2: f64) -> bool {
+    f64::abs(f1 - f2) < f32::EPSILON as f64
+    // Alternative solution:
+    // format!("{:.6}", f1) == format!("{:.6}", f2)
+}
+
+pub fn serde_values_eq(a: &serde_json::Value, b: &serde_json::Value) -> bool {
+    if let (serde_json::Value::Number(n1), serde_json::Value::Number(n2)) = (a, b) {
+        if !n1.is_i64() || !n2.is_i64() {
```

Comment:
This will trigger if one of them isn't an integer. But as I mentioned on discord, if one of them is an integer, then the other one would've been stored properly as an integer too if it was actually the same (even if it passed through a float, up to like a u16 (technically u23, but that doesn't exist)). So if only one is an integer then it should automatically compare false. So it should only go do the if when both are not i64s and then it should automatically fail at the direct comparison.

Reply:
Should this actually matter in practice? Probably not, they should still end up comparing not the same as floats, but I think it might be a bit more correct.

Reply:
Do we have any guarantee that that comment will be kept? Also afaik an integer and a float both get thrown into the json type Number.

Reply:
Hm right, I meant the underlying type used by the server

Reply:
Tbh I think doing it like this is not fine in any case, even if it's technically a float, if you have it in a range from 0-1 if it hits 0 or 1 it'll still be an integer and mess up the condition and the expected nature of the api (+ avoiding making apis that only work properly if you follow additional conditions is like one of the core reasons rust exists :sob:).
Either you check that both are floats and only then do the float codepath (because if one isn't then it should automatically treat it like an integer comparison because data representation wouldn't mess up ints) or you go back to just checking that either of them is a float (this will work, I just think it's slightly less clean).
But the current way is giving me a stroke, I'm sorry.

Reply:
Ok, i'll do the change. Just know that when the value is actually an integer, it will not be mistaken as a float. We had a similar problem in the past and we fixed it, it should be air tight.